test/test_response.py

```diff
@@ -156,79 +161,82 @@ def test_chunked_decoding_gzip(self):
         r = HTTPResponse(fp, headers={'content-encoding': 'gzip'},
                          preload_content=False)
 
-        self.assertEqual(r.read(11), b'')
-        self.assertEqual(r.read(1), b'f')
-        self.assertEqual(r.read(2), b'oo')
-        self.assertEqual(r.read(), b'')
-        self.assertEqual(r.read(), b'')
+        assert r.read(11) == b''
+        assert r.read(1) == b'f'
+        assert r.read(2) == b'oo'
+        assert r.read() == b''
+        assert r.read() == b''
 
     def test_body_blob(self):
         resp = HTTPResponse(b'foo')
-        self.assertEqual(resp.data, b'foo')
-        self.assertTrue(resp.closed)
+        assert resp.data == b'foo'
+        assert resp.closed
 
-    def test_io(self):
+    def test_io(self, sock):
         fp = BytesIO(b'foo')
         resp = HTTPResponse(fp, preload_content=False)
 
-        self.assertEqual(resp.closed, False)
-        self.assertEqual(resp.readable(), True)
-        self.assertEqual(resp.writable(), False)
-        self.assertRaises(IOError, resp.fileno)
+        assert not resp.closed
+        assert resp.readable()
+        assert not resp.writable()
+        with pytest.raises(IOError):
+            resp.fileno()
 
         resp.close()
-        self.assertEqual(resp.closed, True)
+        assert resp.closed
 
         # Try closing with an `httplib.HTTPResponse`, because it has an
         # `isclosed` method.
-        hlr = httplib.HTTPResponse(socket.socket())
+        hlr = httplib.HTTPResponse(sock)
         resp2 = HTTPResponse(hlr, preload_content=False)
-        self.assertEqual(resp2.closed, False)
+        assert not resp2.closed
         resp2.close()
-        self.assertEqual(resp2.closed, True)
+        assert resp2.closed
 
         # also try when only data is present.
         resp3 = HTTPResponse('foodata')
-        self.assertRaises(IOError, resp3.fileno)
+        with pytest.raises(IOError):
+            resp3.fileno()
 
         resp3._fp = 2
         # A corner case where _fp is present but doesn't have `closed`,
         # `isclosed`, or `fileno`.  Unlikely, but possible.
-        self.assertEqual(resp3.closed, True)
-        self.assertRaises(IOError, resp3.fileno)
+        assert resp3.closed
+        with pytest.raises(IOError):
+            resp3.fileno()
 
-    def test_io_closed_consistently(self):
-        hlr = httplib.HTTPResponse(socket.socket())
+    def test_io_closed_consistently(self, sock):
+        hlr = httplib.HTTPResponse(sock)
```

Comment:
This `socket.socket()` object was never closed, so the test would have a `ResourceWarning`. I’ve switched to a fixture so it always gets cleaned up properly.